drop-transfer/src/ws/client/mod.rs

```diff
@@ -456,6 +456,12 @@ async fn start_upload(
         );
 
         let send_file = async {
+            let _permit = state
+                .throttle
+                .acquire()
+                .await
+                .map_err(|_| crate::Error::Canceled)?;
```

Comment:
TBH it can be unwrapped here since the only reason for it to fail is when somebody calls `Semaphore::close()` and nobody is doing that, so really no error here is expected.
But if somebody would call `close()` in the future it will probably be related to the cancelation or something so it's just dumb future-proofing. Definitely not worth introducing yet another error for this IMHO

Reply:
You're right.
A log message would be appreciated though :)